client/lib/url/is-external.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+import { URL as TypedURL } from 'types';
+
+/**
+ * Internal dependencies
+ */
+import { isLegacyRoute } from 'lib/route/legacy-routes';
+
+// Base URL used for URL parsing. The WHATWG URL API doesn't support relative
+// URLs, so we always need to provide a base of some sort.
+const BASE_HOSTNAME = 'base.invalid';
+const BASE_URL = `http://${ BASE_HOSTNAME }`;
+
+export default function isExternal( url: TypedURL ): boolean {
+	// The url passed in might be of form `en.support.wordpress.com`,
+	// so for this function we'll append double-slashes to fake it.
+	// If it is a relative URL the hostname will be the base hostname.
+	if (
+		! url.startsWith( 'http://' ) &&
+		! url.startsWith( 'https://' ) &&
+		! url.startsWith( '/' ) &&
+		! url.startsWith( '?' ) &&
+		! url.startsWith( '#' )
```

Comment:
We're missing the case where the URL already starts with `//`, right?

```suggestion
		! url.startsWith( '#' ) &&
		! startsWith( url, '//' )
```

We don't want to modify the URL to be `////example.com`.

Reply:
Doesn't `url.startsWith( '/' )` already cover that case?

Reply:
🤦‍♂

Reply:
Haha, don't worry, it's easy to miss! 😄